feat(core)!: change dojang default escape|unescape to lodash.template syntax (#7661)
web-infra-dev/rspack: website/docs/en/plugins/rspack/html-rspack-plugin.mdx

:::warning
`rspack.HtmlRspackPlugin` does not support the full `ejs` syntax; it only supports a subset of the `ejs` syntax. If you need full `ejs` syntax support, you can use `html-webpack-plugin` directly.

> In order to align `html-webpack-plugin` default syntax implementation, Rspack change the default `ejs` escape | unescape to be the same as `html-webpack-plugin`'s default syntax.

Reviewer: ```suggestion
In order to align the default template syntax of `html-webpack-plugin`, Rspack changed the default EJS escape and unescape to be the same as `html-webpack-plugin`'s default syntax.
```

:::

> ### Supported `ejs` Syntax

Reviewer: ```suggestion
### Supported EJS Syntax
```

> Only the following basic interpolation expressions are supported. Here, the interpolation expressions only support the most basic string types and do not support arbitrary JavaScript expressions. Other `ejs` syntaxes are currently not supported.

Reviewer: ```suggestion
Only the following basic interpolation expressions are supported. Here, the interpolation expressions only support the most basic string types and do not support arbitrary JavaScript expressions. Other EJS syntaxes are currently not supported.
```

#### \<%-: Escaped output

> Escapes the content within the interpolation

Reviewer: ```suggestion
Escapes the content within the interpolation:
```

```html title="ejs"
<p>Hello, <%- name %>.</p>
<p>Hello, <%- 'the Most Honorable ' + name %>.</p>
```

```json title="locals"
{
  "name": "Rspack<y>"
}
```

```html title="html"
<p>Hello, Rspack&lt;y&gt;.</p>
<p>Hello, the Most Honorable Rspack&lt;y&gt;.</p>
```

#### \<%=: Unescaped output

> Does not escape the content within the interpolation

Reviewer: ```suggestion
Does not escape the content within the interpolation:
```
